### PR TITLE
Defers some list initializations 

### PR DESCRIPTION
**Pull request by AnturK**
Also fixes few missing ..() calls in New(). There's probably more of these.
I don't see measurable difference and the majority of init time lists are still on obj/item and /turf but it doesn't hurt.

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -25,6 +25,7 @@ var/next_mob_id = 0
 	..()
 
 /atom/proc/prepare_huds()
+	hud_list = list()
```
Comment (phil235) [inline code/modules/mob/mob.dm]: not all atom seem to call prepare_huds() on New(), is it going to be ok?

Comment (AnturK) [inline code/modules/mob/mob.dm]: I'll look over these tomorrow, i only did quick "can i see huds at all" test of this so far.

**Comment by MrPerson** [comment]
This would be better off deferring list creation until the lists are actually used. Every single movable and atom does not need a list lying around on the off chance a HUD is added or a mob is buckled to it, especially when most objects can't even be buckled to in the first place.

**Comment by MrStonedOne** [comment]
I should point out, that the overhead comes from that type having **A** var that is initialized to a reference type.

this means that list() shouldn't be your only focus, all reference types like icon() (not the same as single quote icon references, those don't add an init proc) or any datum/atom, initialized in the var's declaration, will cause this same issue.

BUT this also means that majority of the overhead comes from the first reference type initialization, additional initialization don't add that much overhead. The reason for this is because its not the actual initialization that is the overhead, its the proc call of the hidden fake initialization proc.

so if an object has 100 list()s in its var defines, and you remove 99 of them, you haven't done much to cut down on start up time overhead.

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -27,6 +27,9 @@
 		if(user_buckle_mob(M, user))
 			return 1
 
+/atom/movable/New()
+	..()
+	buckled_mobs = list()
```
Comment (MrStonedOne) [inline code/game/objects/buckling.dm]: having to add the New() creates more overhead than you save

**Comment by AnturK** [comment]
@MrStonedOne Yeah, i feared that might be the case, this is going to be less about microoptimization more about refactoring specifc system code toghether anyway.

File: code/game/objects/buckling.dm (hunk 2 above)
Comment (AnturK) [inline code/game/objects/buckling.dm]: This is the point where i realized this optimization is pretty pointless. It had one in lightning already anyway but that file is pretty self-contained. Any idea if the cost of a ..() cloned call is same or lower than new proc by the way ?

Comment (MrStonedOne) [inline code/game/objects/buckling.dm]: Its a bit lower, yes

**Comment by optimumtact** [comment]
lummox recommended leaving the base types null and having a default standin and a proc to check automatically, rather than using new to set them, basically, push the list inits down into the subtypes which are hopefully defined less.

and or intialise/determine the list contents at runtime when they're first used

**Comment by MrPerson** [comment]
If the list is hardcoded, you get pretty good improvements from making the var global instead of giving a different list to every object. Yes this makes badmins cry but they can suck it.

Dunno how much this cuts down on the init proc though. I'm not even sure how to test for that to begin with.

**Comment by coiax** [comment]
Can we just ask lummox to add tuples/static lists to Byond?

**Comment by optimumtact** [comment]
you can make feature rquests on the byond forum if you would like

**Comment by optimumtact** [comment]
conflicts, but are we ready to go when fixed?

**Comment by AnturK** [comment]
Yeah, this is ready. I'll fix the conflicts later today.

**Comment by optimumtact** [comment]
Forgot to say but did you check what @phil235 said?

**Comment by AnturK** [comment]
Yeah, there shouldn't be any issues with these, only non-mob that can have huds (mechs) uses prepare huds right now.

**Pull request merged**